sigmastate/src/main/scala/sigmastate/utxo/transformers.scala

```diff
@@ -69,7 +70,13 @@ case class Append[IV <: SType](input: Value[SCollection[IV]], col2: Value[SColle
     val inputV = input.evalTo[Coll[IV#WrappedType]](env)
     val col2V = col2.evalTo[Coll[IV#WrappedType]](env)
     addSeqCost(Append.costKind, inputV.length + col2V.length) { () =>
-      inputV.append(col2V)
+      if (E.context.currentErgoTreeVersion >= JitActivationVersion) {
```

Comment:
Maybe condition can be done for interpreter version? I see no big semantic problems about it. In case of ergo tree version used, we need for roadmap for compiler and tooling update, but I do not see Github issues or other efforts towards that.

Reply:
If "interpreter version" means release version, then no, the semantics should be connected to ergoTree version, because we never know which release will be executing the ergoTree.

Reply:
Your code is executing a tree, you also know behavior of previous versions. You don't know what future versions will do, but that's their business to preserve compatibility (or not).
If you're going with tree version based versioning, see comment above.

Reply:
I don't understand "You don't know what future versions will do, but that's their business to preserve compatibility (or not)". 
The semantics of each ErgoTree version should be fixed and preserved regardless of which Interpreter implementation executes it. 
Any ErgoTree v0, v1 will still be valid and existing apps can create new transactions with such (not the most recent) versions.
If they opt to not upgrade their code, they will have old semantics and operation behavior. If they need fixed operations they should upgrade. This is their choice.
Appkit already uses the latest activated ergo tree version for new transactions (it takes it from the Node API). 
I don't know about other toolchains, they can use Appkit as reference implementation.

Reply:
I see no any reason to discuss even whether to do documentation for protocol changes or not. 

AppKit is not the only tool and will never be so.

Also, why activation rules for XorOf (proposed in https://github.com/ScorexFoundation/sigmastate-interpreter/pull/777) and Append are different?

Reply:
You are kidding? They were the same, you proposed to switch from versioning based on ergoTree version (the original variant) to versioning based on activatedVersion. I agree with the reasoning there. In principle they can be applied here as well.
Note, there are other fixes in other operations, should we follow the same versioning logic there as well?

I think I described by logic behind the current implementation, don't want to argue on this.

1) You have the final word, let me know whether it should be changed as you suggest or left as is? I will make the changes. 
2) If documentation is necessary, I can describe changes in LangSpec.md

Reply:
Removed changes in Append. The changes will be reintroduced in #771